src/main/java/org/folio/circulation/domain/UpdateLoan.java

```diff
@@ -89,6 +89,8 @@ public UpdateLoan(Clients clients,
     final Loan loan = loanAndRelatedRecords.getLoan();
     LoanPolicy loanPolicy = loan.getLoanPolicy();
 
+    // loanPolicy.recall is a public method and may be called outside of the context of the
+    // loan.wasDueDateChangedByRecall() condition found in the recall method of this class.
```

Comment:
Please can you help understand what the purpose of this comment is?

Reply:
This comment is intended to point out the fact that if loanPolicy.recall is invoked through some other route, aside from `wasDueDateChangedByRecall`, then certain logic would be avoided. Specifically `loan.wasDueDateChangedByRecall()` would not be checked.

Reply:
Maybe this isn't a problem--in fact i do not believe that `loanPolicy.recall` is called in any other context--but it is a potential problem.

Reply:
> Maybe this isn't a problem--in fact i do not believe that `loanPolicy.recall` is called in any other context--but it is a potential problem.

How would you propose this potential problem be mitigated?

Reply:
Outside of introducing a service that can handle the logic for recall in its own private methods, we could add a redundant check for if the loan in question is overdue. I did not want to do this, since it is not needed given the current flow of control. The comment was left in attempt to notify future developers that if `loanPolicy.recall` is called in another context, then certain assumptions (i.e. [that the loan had already been changed via a recall](https://github.com/folio-org/mod-circulation/blob/3c72f15a510a593f19e1a1f32b7f53e25f9c561d/src/main/java/org/folio/circulation/domain/UpdateLoan.java#L72)) may not be valid.